resources/sdk/purecloudjava/tests/ApiClientRetryTest.java

```diff
@@ -0,0 +1,337 @@
+                return null;
+            }
+
+            @Override
+            public Map<String, String> getHeaders() {
+                Map<String, String> headers = new HashMap<>();
+                headers.put("Retry-After", "1");
+                return headers;
+            }
+
+            @Override
+            public boolean hasBody() {
+                return false;
+            }
+
+            @Override
+            public String readBody() throws IOException {
+                return null;
+            }
+
+            @Override
+            public InputStream getBody() throws IOException {
+                return null;
+            }
+
+            @Override
+            public void close() throws Exception {
+
+            }
+        };
+
+        ApiClient.RetryConfiguration retryConfiguration = new ApiClient.RetryConfiguration();
+        retryConfiguration.setMaxRetryTime(30);
+
+        ApiClient.Retry retry = new ApiClient.Retry(retryConfiguration);
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        boolean result = retry.shouldRetry(response);
+        Assert.assertEquals(false, result, "Status Code is 200, so shouldRetry method returns false");
+        Assert.assertTrue(stopwatch.elapsed(TimeUnit.MILLISECONDS) < 100);
+        stopwatch.stop();
+    }
+
+    @Test(priority = 2)
+    public void shouldRetryTestWith_429() throws IOException {
+
+        ApiClientConnectorResponse response = new ApiClientConnectorResponse() {
+            @Override
+            public int getStatusCode() {
+                return 429;
+            }
+
+            @Override
+            public String getStatusReasonPhrase() {
+                return null;
+            }
+
+            @Override
+            public Map<String, String> getHeaders() {
+                Map<String, String> headers = new HashMap<>();
+                headers.put("Retry-After", "1");
+                return headers;
+            }
+
+            @Override
+            public boolean hasBody() {
+                return false;
+            }
+
+            @Override
+            public String readBody() throws IOException {
+                return null;
+            }
+
+            @Override
+            public InputStream getBody() throws IOException {
+                return null;
+            }
+
+            @Override
+            public void close() throws Exception {
+
+            }
+        };
+
+        ApiClient.RetryConfiguration retryConfiguration = new ApiClient.RetryConfiguration();
+        retryConfiguration.setMaxRetryTime(30);
+
+        ApiClient.Retry retry = new ApiClient.Retry(retryConfiguration);
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        boolean result = retry.shouldRetry(response);
+        Assert.assertEquals(true, result, "Status Code is 429, so it will sleep for 1 Sec as provided in Retry-After header and returns true");
+        Assert.assertTrue(stopwatch.elapsed(TimeUnit.MILLISECONDS) < 1100);
```

Comment:
Should we also assert that the elapsed time is over a certain threshold?

Reply:
I think we can do both way but asserting to a calculated time period is more precise than asserting to over a certain threshold..

Reply:
@Tom-J-Barrett yes

@deerghayu per our conversation yesterday, this should be checking for the elapsed time to be within 100ms of the expected runtime.

Reply:
oh I see I what I was missing. thanks guys